### PR TITLE
dfhack: init at 0.40.24-r3

### PR DESCRIPTION
**Pull request by robbinch**
This adds DFHack, a Dwarf Fortress memory access library, distributed with scripts and plugins implementing a wide variety of useful functions and tools.

Run dfhack_install_dir to create a new game directory and play by running ./dfhack in it.

### CODE REVIEW

File: pkgs/games/dfhack/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchgit, dwarf_fortress, cmake, zlib, perl, XMLLibXML, XMLLibXSLT
+}:
+
+let
+  baseVersion = "40";
+  patchVersion = "24-r3";
+  src = fetchgit {
```
Comment (the-kenny) [inline pkgs/games/dfhack/default.nix]: There is ```fetchFromGitHub```

File: pkgs/games/dfhack/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+stdenv.mkDerivation rec {
+  name = "dfhack-0.${baseVersion}.${patchVersion}";
+
+  inherit baseVersion patchVersion src;
+
+  buildInputs = [ cmake zlib perl XMLLibXML XMLLibXSLT ];
+
+  preConfigure = ''
+    export cmakeFlags="-DCMAKE_INSTALL_PREFIX=$prefix/dfhack $cmakeFlags"
```
Comment (the-kenny) [inline pkgs/games/dfhack/default.nix]: You can specify ```cmakeFlags``` as a attribute to ```mkDerivation```

**Comment by robbinch** [comment]
dfhack needs to pull in submodules in compile.  I'm not sure how to specify that it fetchFromGitHub.

I tried cmakeFlags but I do not know how to override -DCMAKE_INSTALL_PREFIX.
I used cmakeFlags = [ "-DCMAKE_INSTALL_PREFIX=$prefix/dfhack" ] but the result was:
cmake flags: -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_NAME_DIR=/nix/store/w24fsnyzn2pf1y0mxm18v7xzc3ksh8fc-dfhack-0.40.24-r3/lib -DCMAKE_INSTALL_PREFIX=/nix/store/w24fsnyzn2pf1y0mxm18v7xzc3ksh8fc-dfhack-0.40.24-r3 -DCMAKE_INSTALL_PREFIX=$prefix/dfhack

I would like files to be installed in /nix/store/...-dfhack-0.40.24-r3/dfhack instead of $out.

**Comment by the-kenny** [comment]
Sorry for leaving this for too long. I figure we can merge this in the current state.

**Comment by the-kenny** [comment]
Merged as 7c149b86c3c2e84a07c7eb575e23c64878507fc8

**Pull request closed without merging**